pkg/crc/api/api_windows.go

```diff
@@ -0,0 +1,74 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+var elog debug.Log
+
+type crcDaemonService struct {
+	socketPath string
+}
+
+func (m *crcDaemonService) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	crcApiServer, err := CreateApiServer(m.socketPath)
+	if err != nil {
+		elog.Error(1, fmt.Sprintf("Failed to start CRC daemon service: %v", err)) // nolint
+		return
+	}
+	// Windows servivce manager calls OnStart when starting the service
+	// and it calls this method, which the service manager expects to handle
+	// commands like stop, shutdown. crcApiServer.Serve is being called as a go routine
+	// since we don't want it block
+	go crcApiServer.Serve()
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+loop:
+	for { // nolint
+		select {
+		case c := <-r:
+			switch c.Cmd {
+			case svc.Interrogate:
+				changes <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				elog.Info(1, "Shutting down CRC daemon service") // nolint
+				break loop
```

Comment:
does this need to be named?

would a function named `MainLoop()` with the `for {};` not work with a `return` instead of a break?

Reply:
@gbraad Updated with the changes you suggested, ptal if that's what you had in mind!